From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Fri, 10 May 2024 18:14:47 +0200
Subject: [PATCH] Fix empty retry_strategy of Batch JobDefinitions causing
 panics

Computing whether a Batch JobDefinition has a diff panics if the
retry_strategy is empty. This is a valid configuration because all
properties of retry_strategy are optional.

This patch fixes that by adding the missing nil checks for the
retry strategy.

diff --git a/internal/service/batch/job_definition.go b/internal/service/batch/job_definition.go
index d1b5102162..6f0d4976e7 100644
--- a/internal/service/batch/job_definition.go
+++ b/internal/service/batch/job_definition.go
@@ -528,12 +528,12 @@ func needsJobDefUpdate(d *schema.ResourceDiff) bool {
 		}
 
 		var ors, nrs *batch.RetryStrategy
-		if len(o.([]interface{})) > 0 {
+		if len(o.([]interface{})) > 0 && o.([]interface{})[0] != nil {
 			oProps := o.([]interface{})[0].(map[string]interface{})
 			ors = expandRetryStrategy(oProps)
 		}
 
-		if len(n.([]interface{})) > 0 {
+		if len(n.([]interface{})) > 0 && n.([]interface{})[0] != nil {
 			nProps := n.([]interface{})[0].(map[string]interface{})
 			nrs = expandRetryStrategy(nProps)
 		}
